File: abc/311/b.cpp
#include <bits/stdc++.h>
#define rep(i, n) for (long long i = 0; i < (long long)(n); ++i)
using namespace std;
using ll = long long;

int main() { 
  ll n, d;
  cin >> n >> d;

  vector<string> s(n);
  rep(i, n) cin >> s[i];

  ll ans = 0;
  ll now = 0;
  rep(j, d) {
    bool ok = true;
    rep(i, n) {
      if (s[i][j] == 'x') ok = false;
    }

    if (ok) ++now;
    else now = 0;

    ans = max(ans, now);
  }

  cout << ans << '\n';
  return 0;
}

File: abc/311/b.rb
N, D = gets.split.map(&:to_i)
S = Array.new(N) {gets}

ans = 0
now = 0

(0..D-1).each do |d|
  ok = true
  S.each do |s|
    ok = false if s[d] == "x"
  end
  
  now = ok ? now + 1 : 0
  ans = [ans, now].max
end

puts ans